### PR TITLE
Add GraphQL documentation

### PR DESCRIPTION
**Pull request by jdrossl**
https://github.com/craftercms/craftercms/issues/2697

### CODE REVIEW

File: source/developers/cook-books/graphql/index.rst [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+3. Obtain the GraphQL schema for your site, you can use the provided GraphiQL client or any third party client
+4. Develop GraphQL queries to use in your site or external app
+
+All content changes made by authors in Crafter Studio will be immediately available in GraphQL queries.
+
+When a change is made in the content model, for example adding a new field or creating a new content-type, the
+GraphQL schema will be rebuilt to reflect the same changes. So for a Crafter CMS site that uses GraphQL queries the
+develop process would look like this:
+
+1. Developers make changes in the content model
```
Comment (sumerjabri) [inline source/developers/cook-books/graphql/index.rst]: Developers define the base content model

File: source/developers/cook-books/graphql/index.rst [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+2. Developers update the GraphQL queries to use the latest schema
```
Comment (sumerjabri) [inline source/developers/cook-books/graphql/index.rst]: Developers define the blueprint/site base GraphQL queries

File: source/developers/cook-books/graphql/index.rst [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+3. Authors make updates using the latest content model
```
Comment (sumerjabri) [inline source/developers/cook-books/graphql/index.rst]: Content authors create content based on the model

File: source/developers/cook-books/graphql/index.rst [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+4. Publishers review & approve the author's work
+5. Publishers publish to live both the content model configuration & the content updates
+6. Crafter Deployer will handle the GraphQL schema rebuild in delivery
+
+You can also use the Crafter CMS GraphQL API from an external site or application, however in this case you will need to
```
Comment (sumerjabri) [inline source/developers/cook-books/graphql/index.rst]: What's this about?

File: source/developers/cook-books/graphql/index.rst [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+1. Create a new site (if needed, for existing sites skip to step number 3)
+2. Define the content model for your site
+3. Obtain the GraphQL schema for your site, you can use the provided GraphiQL client or any third party client
+4. Develop GraphQL queries to use in your site or external app
+
+All content changes made by authors in Crafter Studio will be immediately available in GraphQL queries.
+
+When a change is made in the content model, for example adding a new field or creating a new content-type, the
+GraphQL schema will be rebuilt to reflect the same changes. So for a Crafter CMS site that uses GraphQL queries the
+develop process would look like this:
```
Comment (sumerjabri) [inline source/developers/cook-books/graphql/index.rst]: the development process (not develop)?

File: source/developers/cook-books/graphql/index.rst [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+4. Click ``GraphiQL`` in the left sidebar
+
+To explore the GraphQL schema you can click the ``Docs`` icon on the right side:
+
+.. image:: /_static/images/developer/graphql/graphiql.png
+        :width: 75%
+        :alt: GraphiQL
+        :align: center
+
+GraphiQL provides a convenient search navigation to quickly found a specific type or field:
```
Comment (alhambrav) [inline source/developers/cook-books/graphql/index.rst]: `to quickly found a specific type` -> to quickly **find** a specific type

File: source/developers/cook-books/graphql/index.rst [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+``Website Editorial`` blueprint but the same concepts apply to any Crafter CMS site.
+
+For each content-type in the site you will find a field in the root Query, the name of the field is based on the
+name of the content-type so for ``/page/article`` the field will be ``page_article``.
+These fields contain two sub-fields, one is the ``total`` number of items found by the query and the other is a list
+of ``items``.
+
+.. note::
+  Because GraphQL only supports the underscore ``_`` character besides alphanumeric for names, if your content-type or 
+  field name contains a the dash ``-`` character it will be replaced with a double underscore ``__``. To avoid
```
Comment (alhambrav) [inline source/developers/cook-books/graphql/index.rst]: `field name contains a the dash ``-`` character` -> `field name contains` **a** `dash ``-`` character`

File: source/developers/cook-books/graphql/index.rst [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+        # (names are based on the content-type configuration)
+        title
+        # only return articles from this author
+        author (filter: { matches: "Jane" })
+        date_dt
+      }
+    }
+  }
+
+Finally you can also include fields from child components in your model, this applies for field like ``node-selector``,
```
Comment (alhambrav) [inline source/developers/cook-books/graphql/index.rst]: ` this applies for field like` ->  this applies for field**s** like

File: source/developers/cook-books/graphql/index.rst [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+            # only return articles from this category
+            key (filter: { matches: "health" }) 
+            value_smv
+          }
+        }
+      }
+    }
+  }
+
+You can use many GraphQL features such as aliases and query variables to implement complex queries than can be reused.
```
Comment (alhambrav) [inline source/developers/cook-books/graphql/index.rst]: `complex queries than can be reused` -> complex queries **that** can be reused

File: source/developers/cook-books/graphql/index.rst (hunk 4 above)
Comment (jdrossl) [reply source/developers/cook-books/graphql/index.rst]: Steps 5 & 6 are about publishing the form-definition.xml files to rebuild the schema in delivery.

The last paragraph is about handling when using the API outside of a crafter site

**Pull request merged**